skimage/morphology/_skeletonize.py

```diff
@@ -72,26 +73,26 @@ def skeletonize(image, *, method=None):
            [0, 0, 0, 0, 0, 0, 0, 0, 0]], dtype=uint8)
 
     """
-    image = image.astype(bool, copy=False)
+    image = image.astype(bool, order="C", copy=False)
```

Comment:
Out of curiosity, what prompted this addition? Thanks!

Reply:
I am also curious. 😂

Reply:
All code paths lead eiter to `_compute_thin_image` or `_fast_skeletonize` which both require C-contiguous memoryviews. This makes sure that the function doesn't fail unncessarily when users pass in a different layout. 

While it is currently a bit redundant with the line 
```python
image_o = image.astype(bool, order="C", copy=False)
```
in `_skeletonize_3d`, I thought this doesn't hurt robustness as in "defensive coding".

Reply:
Sure, defensive is better than lax! Thanks for the details. But, in practice, have you run into a failure, or did you just notice the code paths and the current 'risk?'